### PR TITLE
DEV: Correct the ruby version check

### PR DESCRIPTION
**Pull request by tisonkun**
also fix discourse cannot work with ruby 3.0.0

### CODE REVIEW

**Comment by TisonKun** [comment]
for totally compatible with ruby 3.0.0 we still need to fix issues such as [this one](https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/).

but change the way we compare semver is always a net win.

**Review by CvX — APPROVED** [review]

File: config/application.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # note, we require 2.5.2 and up cause 2.5.1 had some mail bugs we no longer
 # monkey patch, so this avoids people booting with this problem version
 begin
-  if !RUBY_VERSION.match?(/^2\.(([67])|(5\.[2-9]))/)
+  unless Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("2.5.2")
```
Comment (CvX) [inline config/application.rb:6]:
```suggestion
  if Gem::Version.new(RUBY_VERSION) < Gem::Version.new("2.5.2")
```

**Review by CvX — APPROVED** [review]

**Comment by CvX** [comment]
Thank you! 😃

**Comment by tisonkun** [comment]
@CvX thanks for your review! I'll try to run discourse on Ruby 3.0.0 later :-)

**Pull request merged**